### PR TITLE
skip undefined

### LINKED ISSUE #6: recent release not safe for undefined
We're getting exceptions from our clients using the heap integration trying to call toString() on undefined. The following code switched from checking for null or undefined to just checking for null:

https://github.com/segment-integrations/analytics.js-integration-heap/pull/5/files#diff-6d186b954a58d5bb740f73d84fe39073R107

I think you'd want to switch back to `==` or use `value === null || value === undefined`

### PR DESCRIPTION
**Pull request by hankim813**
fixes https://github.com/segment-integrations/analytics.js-integration-heap/issues/6

### CODE REVIEW

File: test/index.test.js [hunk 1]
```diff
@@ -158,6 +158,11 @@ describe('Heap', function() {
         analytics.called(window.heap.track, 'event', { property: true });
       });
 
+      it('should send undefined or null props', function() {
```
Comment (ndhoule) [inline test/index.test.js]: `should filter out undefined and null properties`?

File: lib/index.js [hunk 2]
```diff
@@ -104,7 +104,8 @@ function clean(obj) {
   for (var k in obj) {
     if (obj.hasOwnProperty(k)) {
       var value = obj[k];
-      if (value === null) continue;
+      // don't send null or undefined values
```
Comment (ndhoule) [inline lib/index.js]: The code is self-explanatory; instead explain _why_ we're doing this. (e.g. `null and undefined properties cause Heap to reject the message, filter them out`)

Comment (ndhoule) [inline lib/index.js]: As a general rule: Comments should only ever describe _what_ the code is doing when the code is very complex. Figuring out what a line of code is doing without help is usually pretty easy. Figuring out the context behind that code (the _why_) is usually difficult or impossible; your comments should aspire to answer this question

Comment (hankim813) [inline lib/index.js]: Yep! Totally make sense. Thanks @ndhoule !

**Pull request merged**